README.md

```diff
@@ -2,35 +2,65 @@
 
-PRs and feedback are welcomed. A good place to start is to review the current [open issues](https://github.com/DemocracyEarth/contracts/issues), please feel free to engage with us by commenting in a specific issue. Remember you can also check out the [Waffle board](https://waffle.io/DemocracyEarth/contracts) to understand what's being worked on now and what is coming up next.
+## Contributing
 
-## How to test locally
+PRs and feedback are welcomed. A good place to start is to review the current [open issues](https://github.com/DemocracyEarth/dapp/issues), please feel free to engage with us by commenting in a specific issue. Remember you can also check out the [Waffle board](https://waffle.io/DemocracyEarth/contracts) to understand what's being worked on now and what is coming up next.
 
-Project is being developed with NodeJs and Truffle, please make sure you have the latest version of npm before continuing. First follow the instructions to install [Truffle](http://truffleframework.com/):
+Dapp development is discussed actively in the slack channel [#developers](https://democracyearth.slack.com/messages/C5KCH0PD1) in democracyearth.slack.com 
 
-```sh
-$ npm install -g truffle
-```
+The dapp consists of mailnly two elements, contracts and frontend, located respectively in the following folders:
+- **contracts**: Solidity contracts, to be deployed to a local or remote blockchain.
+- **dApp-Srv-Exp**: JS frontend built using npm, bower and http-server
 
-Then clone or fork the repo, do an `npm install`, and you can start running the tests. You will need to run `truffle develop` to start a testing blockchain and then `test` to execute testing files. Please read the [Truffle docs](http://truffleframework.com/docs/getting_started/testing) to learn more about the tool.
+## Development dependencies
 
-## Compile contracts
+- NodeJS (make sure you have the latest version of npm, using `npm i -g npm`)
+  - Bower (`npm install -g bower`)
+  - Truffle (`npm install -g truffle`)
+  - http-server (`npm install -g http-server`)
+- Ganache ([http://truffleframework.com/ganache/]())
+- Metamask browser extension ([https://metamask.io/]())
 
-```sh
-truffle compile
-```
+## How to run
 
-## How to deploy the contracts
+### Compile, deploy and copy compiled contracts to frontend
+
+Choose the environment.
+Use *development* for local environment (start ganache before) or *ropsten* for Ropsten Ethereum test network in remote environment.
+
+ ```sh
+ $ npm install
+ $ truffle compile
+ $ truffle migrate --network [development / ropsten]
+ $ cp build/contracts/* dApp-Svr-Exp/contracts/
+ ```
+ 
+### Start frontend
 
 ```sh
-$ truffle migrate --network ropsten
-cp build/contracts/* dApp-Svr-Exp/contracts/
+$ http-server dApp-Svr-Exp/
```

Comment:
`npm start` will also do the trick

Reply:
And if user does `npm start` she still needs the `http-server` package installed though, correct?

Reply:
@LucasIsasmendi are we sure the above considers all the steps? I am setting up the dapp repo on my new mac, and after `truffle compile` I get this error:

![image](https://user-images.githubusercontent.com/7900773/41249678-a3ce21cc-6dac-11e8-85c6-b16cd62e6fbe.png)

Reply:
you need to pull again the branch, the code shown there is not up to date

Reply:
@aecc I was on latest version of master for the above. Switched to branch `ballot-dashboard` and the same error.

Reply:
![image](https://user-images.githubusercontent.com/7900773/41250566-3c3dc226-6daf-11e8-903a-6a452d79cbdc.png)

Reply:
Then you have something cached, try to clean you cache or the build/ folder. Check the LiquidDemocracy.sol in this PR and you will see the pragma line is different.